jboss/container/wildfly/launch/jgroups/added/launch/jgroups.sh

```diff
@@ -331,20 +333,34 @@ configure_jgroups_encryption() {
   ;;
   "ASYM_ENCRYPT")
       log_info "Configuring JGroups cluster traffic encryption protocol to ASYM_ENCRYPT."
-      if [ -n "${JGROUPS_ENCRYPT_SECRET}"      -o \
-           -n "${JGROUPS_ENCRYPT_NAME}"         -o \
-           -n "${JGROUPS_ENCRYPT_PASSWORD}"     -o \
-           -n "${JGROUPS_ENCRYPT_KEYSTORE_DIR}" -o \
-           -n "${JGROUPS_ENCRYPT_KEYSTORE}" ] ; then
-        log_warning "The specified JGroups configuration properties (JGROUPS_ENCRYPT_SECRET, JGROUPS_ENCRYPT_NAME, JGROUPS_ENCRYPT_PASSWORD, JGROUPS_ENCRYPT_KEYSTORE_DIR JGROUPS_ENCRYPT_KEYSTORE) will be ignored when using JGROUPS_ENCRYPT_PROTOCOL=ASYM_ENCRYPT. Only JGROUPS_CLUSTER_PASSWORD is used."
-      fi
 
-      # CLOUD-2437 AUTH protocol is required when using ASYM_ENCRYPT protocol: https://github.com/belaban/JGroups/blob/master/conf/asym-encrypt.xml#L23
+      # CLOUD-2437 AUTH protocol is required when using ASYM_ENCRYPT protocol
       if [ -n "${JGROUPS_CLUSTER_PASSWORD}" ]; then
-        if [ "${encrypt_conf_mode}" = "xml" ]; then
-          jgroups_encrypt=$(create_jgroups_encrypt_asym)
-        elif [ "${encrypt_conf_mode}" = "cli" ]; then
-          jgroups_encrypt=$(create_jgroups_encrypt_asym_cli)
+        # Test if we have all the information available to configure ASYM_ENCRYPT protocol configuring the Key Store in the Elytron subsystem
+        valid_state=$(validate_keystore "${JGROUPS_ENCRYPT_SECRET}" "${JGROUPS_ENCRYPT_NAME}" "${JGROUPS_ENCRYPT_PASSWORD}" "${JGROUPS_ENCRYPT_KEYSTORE}")
+
+        if [ "${has_elytron_subsystem}" -eq 0 ] && [ "${valid_state}" = "valid" ]; then
+          log_info "Detected valid JGroups encryption configuration, the communication within the cluster will be encrypted using ASYM_ENCRYPT and Elytron keystore."
+          if [ "${key_store_conf_mode}" = "xml" ]; then
+            validate_keystore_and_create "xml" "ASYM_ENCRYPT"
+          elif [ "${key_store_conf_mode}" = "cli" ]; then
+            validate_keystore_and_create "cli" "ASYM_ENCRYPT"
+          fi
+        else
+
+          if [ ! "${has_elytron_subsystem}" -eq 0 ]; then
+            log_warning "Elytron subsystem is not in your configuration, the communication within the cluster will be encrypted using a deprecated version of ASYM_ENCRYPT protocol."
+          elif [ -n "${JGROUPS_ENCRYPT_SECRET}" ] || [ -n "${JGROUPS_ENCRYPT_NAME}" ] || [ -n "${JGROUPS_ENCRYPT_PASSWORD}" ] || [ -n "${JGROUPS_ENCRYPT_KEYSTORE}" ] || [ -n "${JGROUPS_ENCRYPT_KEYSTORE_DIR}" ]; then
+            log_warning "Detected partial JGroups encryption configuration, the communication within the cluster will be encrypted using a deprecated version of ASYM_ENCRYPT protocol. You need to set all of these variables to configure ASYM_ENCRYPT using the Elytron keysore: JGROUPS_ENCRYPT_SECRET, JGROUPS_ENCRYPT_NAME, JGROUPS_ENCRYPT_PASSWORD, JGROUPS_ENCRYPT_KEYSTORE."
```

Comment:
I am wandering if you could have a local var with the list of missing env vars. This info is shared between 2 cases. It would be easy to fix one message and forget about the other one.

Reply:
Correct, fixed, thanks